### PR TITLE
HDDS-7108. Include supported EC data-parity combinations in error message

### PR DESCRIPTION
**Pull request by DaveTeng0**
## What changes were proposed in this pull request?

Need proper error message on console output when incorrect replication data-parity config is passed

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-7108

## How was this patch tested?

Manually test from developer machine

### CODE REVIEW

**Comment by DaveTeng0** [comment]
@umamaheswararao @kerneltime  Feel free to take a look if the added message helps! Thanks!

**Comment by kerneltime** [comment]
LGTM, please address failing CI tests

**Comment by DaveTeng0** [comment]
> LGTM, please address failing CI tests

sure! looking

**Review by kerneltime — APPROVED** [review]

**Review by kerneltime — APPROVED** [review]

**Comment by kerneltime** [comment]
Thank you, @DaveTeng0 for this patch.

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/client/ReplicationConfigValidator.java [hunk 1]
```diff
@@ -61,9 +61,11 @@ public ReplicationConfig validate(ReplicationConfig replicationConfig) {
...
               "-" + ecConfig.getParity() + "-{CHUNK_SIZE}";
         }
-        throw new IllegalArgumentException("Invalid replication config " +
-            "for type " + replicationConfig.getReplicationType() +
-            " and replication " + replication);
+        throw new IllegalArgumentException(
+                "Invalid data-parity replication config " +
+                        "for type " + replicationConfig.getReplicationType() +
+                        " and replication " + replication + "." +
+                        " Supported data-parity are 3-2,6-3,10-4");
```
Comment (adoroszlai) [inline hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/client/ReplicationConfigValidator.java:68]: Sorry for the late review.

This exception may be thrown for non-EC replication config, too, in which case `data-parity` is not applicable.  The message should be different based on replication type.

CLI commands validate RATIS replication factor before reaching this check, but Java code using `RpcClient#createKey` directly could encounter this problem.

**Pull request merged**